source/mod_particles.f90

```diff
@@ -146,6 +159,11 @@ subroutine part_updatePartEnergy(refArray)
   omoidpsv(1:napx) = ((one-mtc(1:napx))*oidpsv(1:napx))*c1e3
   rvv(1:napx)      = (ejv(1:napx)*e0f)/(e0*ejfv(1:napx))     ! Beta_0 / beta(j)
 
+  if(doUpdateAngle) then ! Update particle angle, but only if energy change during tracking
+    yv1(1:napx)    = (ejf0v(1:napx)/ejfv(1:napx))*yv1(1:napx)
+    yv2(1:napx)    = (ejf0v(1:napx)/ejfv(1:napx))*yv2(1:napx)
+  end if
+
```

Comment:
comment in this subroutine are a bit misleading - you update in the subroutine particle angles if explicitly requested, but the request in general is done only if the energy changes during tracking, i.e. your flag here does not remind that it is true only when energy is changed - and strictly speaking there is nothing checking that this is true

Reply:
The comment is just outdated. There used to be a second element in that if statement. Will update.